### PR TITLE
delete err when return _

### PR DESCRIPTION
**Pull request by yu8833**
Signed-off-by: yupengzte <yu.peng36@zte.com.cn>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by cblecker** [comment]
What is the need for this? It looks like two lines down, it calls for err, so I'm not sure why we wouldn't want to return it.

**Review by janetkuo — COMMENTED** [review]
`getAllDaemonSetPods` will always return `nil` error. We don't need to return error in `getAllDaemonSetPods` at all.

**Comment by janetkuo** [comment]
@k8s-bot ok to test

**Comment by YuPengZTE** [comment]
@k8s-bot ok to test

File: pkg/controller/daemon/update.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func (dsc *DaemonSetsController) rollingUpdate(ds *extensions.DaemonSet) error {
 		return fmt.Errorf("couldn't get node to daemon pod mapping for daemon set %q: %v", ds.Name, err)
 	}
 
-	_, oldPods, err := dsc.getAllDaemonSetPods(ds, nodeToDaemonPods)
```
Comment (lukaszo) [inline pkg/controller/daemon/update.go]: If getAllDaemonSetPods is always returning `nil`  just change the function itself to not return an error.

Comment (YuPengZTE) [inline pkg/controller/daemon/update.go]: OK, good idea.
Changes have done, PTAL.
Thanks @lukaszo

**Review by janetkuo — APPROVED** [review]

**Pull request merged**